app/components/quiz/QuizContext.js

```diff
@@ -0,0 +1,43 @@
+import React from 'react';
+import AppStore from 'store/appStore';
+
+export default React.createClass({
+  displayName: 'QuizContext',
+
+  contextTypes: {
+    router: React.PropTypes.func
+  },
+
+  propTypes: {
+    context: React.PropTypes.object.isRequired
+  },
+
+  render() {
+    var text = this.props.context.text;
+    var highlights = this.props.context.highlights;
+
+    var start = 0;
```

Comment:
it takes care of if there's a tail after the highlighted portions which needs to be displayed i.e.

`highlights = [ [0,5] ]
text = "01234this is a tail which is not highlighted so it will be shown in a different color"`